src/core/ddsi/src/ddsi_typelib.c

```diff
@@ -132,15 +132,13 @@ void ddsi_typeinfo_fini (ddsi_typeinfo_t *typeinfo)
 
 const ddsi_typeid_t *ddsi_typeinfo_minimal_typeid (const ddsi_typeinfo_t *typeinfo)
 {
-  DDSRT_STATIC_ASSERT (offsetof (struct ddsi_typeid, x) == 0);
```

Comment:
Why did you remove these? I might be that a comment is missing somewhere, but IIRC and also looking around a bit, the idea was that ensure converting a `ddsi_typeid_t *` pointer into a `DDS_XTypes_TypeIdentifier *` works, and that this is just a sanity check. (Also in ddsi_typewrap.c)

Maybe something changed, or I am overlooking something. That's quite possible, too!

Reply:
I noticed that a few static asserts were missing, so I've added them all in [ddsi_xt_impl.h](https://github.com/eclipse-cyclonedds/cyclonedds/blob/26e670521cc78f872181a8b9369e3563ffc8baa5/src/core/ddsi/include/dds/ddsi/ddsi_xt_impl.h#L290)